src/utils.js

```diff
@@ -287,6 +303,40 @@ genotet.utils.keySet = function(collection) {
 };
 
 /**
+ * Validates email address.
+ * @param {string} email
+ * @return {boolean}
+ */
+genotet.utils.validateEmail = function(email) {
```

Comment:
Can't you read my message more carefully and get the essense? What I'm emphasizing is that you need to combine those "validate" functions into one because they have exactly the same structure and are quite redundant in shape.